### ISSUE TITLE
Documentation: Specifiy minimal version of junit

### ISSUE THREAD

**Issue by FrankR85** [jlink/jqwik#95]
jqwik doesn't work out-of-the-box with the JUnit5 (1.5.2) that comes with SpringBoot 2.2.4.

Running jqwik tests give:
```
WARNUNG: TestEngine with ID 'jqwik' failed to discover tests
java.lang.NoSuchMethodError: org.junit.platform.engine.EngineDiscoveryRequest.getDiscoveryListener()Lorg/junit/platform/engine/EngineDiscoveryListener;
	at net.jqwik.engine.discovery.JqwikDiscoverer.discover(JqwikDiscoverer.java:30)
	at net.jqwik.engine.JqwikTestEngine.discover(JqwikTestEngine.java:50)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverEngineRoot(DefaultLauncher.java:130)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverRoot(DefaultLauncher.java:117)
	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:82)
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestReference.<init>(JUnit5TestReference.java:42)
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestLoader.createUnfilteredTest(JUnit5TestLoader.java:73)
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestLoader.createTest(JUnit5TestLoader.java:63)
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestLoader.loadTests(JUnit5TestLoader.java:50)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:522)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:760)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:460)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:206)
```

Upgrading to 
```
		<dependency>
			<groupId>org.junit.platform</groupId>
			<artifactId>junit-platform-engine</artifactId>
			<version>1.6.1</version>
			<scope>test</scope>
		</dependency>
```
fixes the issue.

The doc should specify the required version for `junit`.

**Comment by jlink** [comment]
That's a good idea and I will add it to the documentation.

BTW, the minimal version is documented in jqwik's Spring support: https://github.com/jlink/jqwik-spring#supported-junit-platform-versions

**Comment by luvarqpp** [comment]
Sidenote: a few days ago was released spring boot milestone 2.3.0.M4, which has bumped to junit-platform 1.6.1.

**Comment by jlink** [comment]
Added required JUnit platform version to current user guide and to template for future user guides: https://jqwik.net/docs/current/user-guide.html#required-version-of-junit-platform
